tools/src/main/java/org/apache/kafka/trogdor/workload/SustainedConnectionWorker.java

```diff
@@ -0,0 +1,563 @@
+                SustainedConnectionWorker.log.error("Abort while maintaining sustained connections", e.getMessage());
+            }
+        }
+    }
+
+    private Optional<SustainedConnection> findConnectionToMaintain() {
+        synchronized (SustainedConnectionWorker.this) {
+            for (SustainedConnection connection : this.connections) {
+                if (connection.needsRefresh()) {
+                    connection.claim();
+                    return Optional.of(connection);
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    private class StatusUpdater implements Runnable {
+        @Override
+        public void run() {
+            try {
+                JsonNode node = JsonUtil.JSON_SERDE.valueToTree(
+                    new StatusData(
+                            SustainedConnectionWorker.this.totalProducerConnections.get(),
+                            SustainedConnectionWorker.this.totalProducerFailedConnections.get(),
+                            SustainedConnectionWorker.this.totalConsumerConnections.get(),
+                            SustainedConnectionWorker.this.totalConsumerFailedConnections.get(),
+                            SustainedConnectionWorker.this.totalMetadataConnections.get(),
+                            SustainedConnectionWorker.this.totalMetadataFailedConnections.get(),
+                            SustainedConnectionWorker.this.totalAborts.get(),
+                            System.currentTimeMillis()));
+                status.update(node);
+            } catch (Exception e) {
+                SustainedConnectionWorker.log.error("Abort while running StatusUpdater", e.getMessage());
+                WorkerUtils.abort(log, "StatusUpdater", e, doneFuture);
+            }
+        }
+    }
+
+    public static class StatusData {
+        private final long totalProducerConnections;
+        private final long totalProducerFailedConnections;
+        private final long totalConsumerConnections;
+        private final long totalConsumerFailedConnections;
+        private final long totalMetadataConnections;
+        private final long totalMetadataFailedConnections;
+        private final long totalAborts;
+        private final long updatedMs;
+
+        @JsonCreator
+        StatusData(@JsonProperty("totalProducerConnections") long totalProducerConnections,
+                   @JsonProperty("totalProducerFailedConnections") long totalProducerFailedConnections,
+                   @JsonProperty("totalConsumerConnections") long totalConsumerConnections,
+                   @JsonProperty("totalConsumerFailedConnections") long totalConsumerFailedConnections,
+                   @JsonProperty("totalMetadataConnections") long totalMetadataConnections,
+                   @JsonProperty("totalMetadataFailedConnections") long totalMetadataFailedConnections,
+                   @JsonProperty("totalAborts") long totalAborts,
```

Comment:
Does it make sense to call this `abortedThreads`?

Reply:
I called it totalAbortedThreads.